core/trino-main/src/main/java/io/trino/sql/analyzer/Field.java

```diff
@@ -13,18 +13,18 @@
  */
 package io.trino.sql.analyzer;
 
-import io.trino.metadata.QualifiedObjectName;
+import com.google.common.collect.ImmutableList;
 import io.trino.spi.type.Type;
 import io.trino.sql.tree.QualifiedName;
 
+import java.util.List;
 import java.util.Optional;
 
 import static java.util.Objects.requireNonNull;
 
 public class Field
 {
-    private final Optional<QualifiedObjectName> originTable;
-    private final Optional<String> originColumnName;
+    private final List<OriginColumnDetail> originColumnDetails;
```

Comment:
Could you explain how it is possible to have more than 1 element in this list?

Reply:
Like if we have field for a expression like `func(func2(col1, col2), col3)` then we might need fetch the `OriginColumnDetail` for `col1`, `col2` and `col3`